src/shared_modules/utils/tests/reflectiveJson_test.cpp

```diff
@@ -0,0 +1,218 @@
+    std::string result = jsonFieldToString(map);
+
+    EXPECT_EQ(result, "{\"outerKey\":{\"innerKey\":123}}");
+}
+
+TEST_F(ReflectiveJsonTest, EscapedStringValue)
+{
+    std::unordered_map<std::string, std::string> map {{"key", "He said \"Hello\""}};
+
+    std::string result = jsonFieldToString(map);
+
+    EXPECT_NE(result.find(R"("key":"He said \"Hello\"")"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BasicStructSerialization)
+{
+    TestData<float> obj;
+    obj.fieldOne = "001";
+    obj.fieldTwo = 30;
+    obj.fieldThree = 1.3;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"001")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":30)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":1.3)"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, NestedStructSerialization)
+{
+    struct NestedData
+    {
+        std::string innerField;
+        int innerValue;
+
+        REFLECTABLE(MAKE_FIELD("innerField", &NestedData::innerField),
+                    MAKE_FIELD("innerValue", &NestedData::innerValue));
+    };
+
+    TestData<NestedData> obj;
+    obj.fieldOne = "Outer";
+    obj.fieldTwo = 2;
+    obj.fieldThree.innerField = "Inner";
+    obj.fieldThree.innerValue = 42;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"Outer")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":{"innerField":"Inner","innerValue":42})"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BasicStructSerializationWithEmptyFields)
+{
+    TestData<std::string_view> obj;
+    obj.fieldOne = "";
+    obj.fieldTwo = 0;
+    obj.fieldThree = "";
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_EQ(json.find(R"("fieldOne":)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":0)"), std::string::npos);
+    EXPECT_EQ(json.find(R"("fieldThree":)"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BasicStructSerializationWithQuotes)
+{
+    TestData<float> obj;
+    obj.fieldOne = "\"001\"";
+    obj.fieldTwo = 30;
+    obj.fieldThree = 1.3;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"\"001\"")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":30)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":1.3)"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, BooleanValues)
+{
+    TestData<bool> obj;
+    obj.fieldOne = "\"001\"";
+    obj.fieldTwo = 30;
+    obj.fieldThree = true;
+
+    std::string json;
+    serializeToJSON(obj, json);
+
+    EXPECT_NE(json.find(R"("fieldOne":"\"001\"")"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldTwo":30)"), std::string::npos);
+    EXPECT_NE(json.find(R"("fieldThree":true)"), std::string::npos);
+}
+
+TEST_F(ReflectiveJsonTest, SpecialCharactersInKeysAndValues)
+{
+    std::unordered_map<std::string, std::string> map {{"newline", "Line1\nLine2"},
```

Comment:
I would add more special characters to test the full ESCAPE_TABLE.
We are missing the non-printable characters that require UNICODE.

Reply:
Good catch, I adding now